src/main/java/org/testng/DependencyMap.java

```diff
@@ -21,7 +22,20 @@
 
   public DependencyMap(ITestNGMethod[] methods) {
     for (ITestNGMethod m : methods) {
-    	m_dependencies.put( m.getRealClass().getName() + "." +  m.getMethodName(), m);
+      m_dependencies.put( m.getRealClass().getName() + "." +  m.getMethodName(), m);
```

Comment:
`MethodHelper` is updated as well but dependencies are validated again in `DependencyMap`, so had to change in 2 places.

I was also thinking about the following: 
Currently if we have:
```
class ClassA {
 @Test(dependsOnMethods="methodB") public void methodA(){}
 @Test public void methodB(){}
}

class ClassB extends ClassA {
 @Test public void methodB(){}
}
```
When we run ClassB tests method ITestNGMethod([ClassA.methodA]).getMethodsDependedUpon() returns "ClassA.methodB". So I was thinking maybe getMethodsDependedUpon() should return "ClassB.methodB" as it is the real implementation which would be used when we run ClassB. I've checked BaseTestMethod.initGroups() lines 469-477 where value of ITestNGMethod.m_methodsDependedUpon is set but in this place in the code there is no way to know in which exact instance test is invoked, only references to ClassA exist.

What do you think about this?